### PR TITLE
WIP: adds conversion from TimeArray -> AxisArray, and various other methods

### PR DESCRIPTION
**Pull request by milktrader**
Abandoning the brainstorm to use nested modules and taking @tshort approach to using `@require`.

This branch will provide the ability to experiment with time-flavored AxisArrays and begins work on some related methods

- [x] `merge`
- [x] `moving` (might name this differently)

Also, the tests are not part of the overall test suite but ensure that conversions are happening correctly.

### CODE REVIEW

File: src/timeseries.jl [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# these methods are not specific to time series necessarily, but quarantined here for the time being
+
+import TimeSeries.moving, Base.merge
+
+# only works for single column AxisArrays
+function moving(a::AxisArray, f::Function, n::Int)
+    m = length(a)-n+1
+    res = zeros(m)
+    for i in 1:m
+        res[i] = f(a[Interval(a.axes[1][i],a.axes[1][i+n-1])])
```
Comment (mbauman) [inline src/timeseries.jl]: This can be expressed without the `Interval` — simply `f(a[Axis{1}(i:i+n-1)])`

**Comment by mbauman** [comment]
Awesome.  I've been thinking a bit about this.  Here's how the moving function might look if it were generalized to work over any dimension:

```julia
function moving{Ax<:Axis}(a::AxisArray, f::Function, n::Ax)
    v = n.val
    m = size(a, n) - v
    res = similar(a, (Ax(m),), (axes(a, Ax)[cld(v,2):end-fld(v,2)],))
    for i in 1:m
        res[Ax(i)] = f(a[Ax[i:i+v-1])
    end
end
```

You'd call it like this: `moving(AAPL, mean, Axis{:time}(10))` for a 10-day moving average across the time axis.

Of course, this requires my custom indexing patch, as well as a specialized `similar` function that I've not written yet.

File: src/timeseries.jl (hunk 1 above)
Comment (milktrader) [inline src/timeseries.jl]: Much nicer.

Comment (milktrader) [inline src/timeseries.jl]: The refactor passes ...

````julia
julia> @test AxisArrays timeseries
conversion from TimeArray to AxisArray
     - axes values are correct
     - data values are correct
5 facts verified.
moving
     - moving only works on single column
2 facts verified.
merge
     - only inner join supported
1 fact verified.
````

**Comment by mbauman** [comment]
It's very interesting to me how close that general method is to allowing image processing here… you could easily extend it to allow an arbitrary number of axes to define a moving square or cube through your data.  If we used `mean!`/`median!` then the indexing shapes themselves would define the reduction dimensions.

Of course the most performant way to do a running mean would be through filtering or fft techniques (as a convolution of a weighted vector).  And there are better methods for running median, too.

**Comment by mbauman** [comment]
I'm going to close this for now; it's out of date and I think we need to move away from `@require` method of lazy dependencies. Feel free to resurrect if you ever are interested in using AxisArrays. Thanks for helping me brainstorm how this package works!

**Comment by milktrader** [comment]
Good call.

**Pull request closed without merging**